realm/realm-library/src/androidTest/java/io/realm/internal/CollectionTests.java

```diff
@@ -396,15 +398,16 @@ public void onChange(Collection collection1) {
                         assertEquals(collection1.size(), 4);
                         break;
                     case 1:
+                        assertFalse(transactionCommitted.get());
                         assertEquals(collection1.size(), 5);
                         sharedRealm.close();
-                        looperThread.testComplete();
                         break;
                 }
             }
         });
         addRow(sharedRealm);
-        assertEquals(collection.size(), 5);
+        transactionCommitted.set(true);
+        looperThread.testComplete();
```

Comment:
This seems to change the flow of this test. Before it was depending on the notification arriving through the looper, now it is triggered synchronously. 

Wouldn't it be better to make a new test case `listenerTriggeredOnCommit()` or something that tests that the change listener is triggered _both_ on `beginTransaction()` and `commitTransaction()` ?